### PR TITLE
jsx-boolean-value.js documentation

### PR DESCRIPTION
**Pull request by caroline223**
Attempted to define the data types of parameters expressed within functions within the jsx-boolean-value.js file.

### CODE REVIEW

**Comment by caroline223** [comment]
@ljharb Please review my PR.

File: package.json [hunk 1]
```diff
@@ -52,10 +52,11 @@
     "@typescript-eslint/parser": "^2.34.0 || ^3.10.1 || ^4.0.0 || ^5.0.0",
     "aud": "^2.0.0",
     "babel-eslint": "^8 || ^9 || ^10.1.0",
-    "eslint": "^3 || ^4 || ^5 || ^6 || ^7 || ^8",
+    "eslint": "^8.18.0",
```
Comment (ljharb) [inline package.json:55]: This will break things; it needs to be reverted.

File: package.json [hunk 2]
```diff
@@ -52,10 +52,11 @@
...
     "eslint-config-airbnb-base": "^15.0.0",
     "eslint-plugin-eslint-plugin": "^2.3.0 || ^3.5.3 || ^4.0.1",
     "eslint-plugin-import": "^2.26.0",
+    "eslint-plugin-react": "^7.30.1",
```
Comment (ljharb) [inline package.json:59]: I’m not sure why this is here?

File: lib/rules/jsx-boolean-value.js [hunk 3]
```diff
@@ -30,15 +34,25 @@ function getErrorData(exceptions) {
...
 function isAlways(configuration, exceptions, propName) {
   const isException = exceptions.has(propName);
   if (configuration === ALWAYS) {
     return !isException;
   }
   return isException;
 }
-
+/**
+     * @param {string} configuration
```
Comment (ljharb) [inline lib/rules/jsx-boolean-value.js:51]: all these blocks need to be deindented a bit

**Review by ljharb — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (caroline223) [reply package.json]: Done. @ljharb

File: package.json (hunk 2 above)
Comment (caroline223) [reply package.json]: Done. @ljharb

File: lib/rules/jsx-boolean-value.js (hunk 3 above)
Comment (caroline223) [reply lib/rules/jsx-boolean-value.js]: Done. @ljharb

**Review by ljharb — APPROVED** [review]

**Pull request merged**